misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/OptionalAssertionsSpec.kt

```diff
@@ -1,45 +1,85 @@
+    isPresentNullable: Fun1<Optional<Int?>, Expect<Int?>.() -> Unit>,
     describePrefix: String = "[Atrium] "
 ) : Spek({
 
-    include(object : SubjectLessSpec<Optional<Any>>(
+    include(object : SubjectLessSpec<Optional<Int>>(
         "$describePrefix[Path] ",
-        isEmpty.forSubjectLess()
+        isEmpty.forSubjectLess(),
+        isPresentFeature.forSubjectLess().adjustName { "$it feature" },
+        isPresent.forSubjectLess { toBe(1) }
+    ) {})
+    include(object : SubjectLessSpec<Optional<Int?>>(
+        describePrefix,
+        isPresentNullableFeature.forSubjectLess().adjustName { "$it feature" },
+        isPresentNullable.forSubjectLess { toBe(1) }
+    ) {})
+    include(object : AssertionCreatorSpec<Optional<Int>>(
+        describePrefix, Optional.of(2),
+        isPresent.forAssertionCreatorSpec("$toBeDescr: 2") { toBe(2) }
+    ) {})
+    include(object : AssertionCreatorSpec<Optional<Int?>>(
+        describePrefix, Optional.of(2) as Optional<Int?>,
+        isPresentNullable.forAssertionCreatorSpec("$toBeDescr: 2") { toBe(2) }
     ) {})
 
     fun describeFun(vararg funName: String, body: Suite.() -> Unit) =
         describeFunTemplate(describePrefix, funName, body = body)
 
-    describeFun(isEmpty.name) {
+    describeFun(isEmpty.name, isPresent.name) {
         val isEmptyFun = isEmpty.lambda
+        val isPresentFun = isPresentFeature.lambda
 
         context("is empty") {
-            it ("does not throw") {
-                val emptyValue = Optional.empty<Any>()
+            it("does not throw") {
+                val emptyValue = Optional.empty<Int>()
                 expect(emptyValue).isEmptyFun()
             }
 
-            it ("throws an AssertionError") {
-                val nonEmptyValue: Optional<Any> = Optional.of(2)
+            it("throws an AssertionError") {
+                val nonEmptyValue: Optional<Int> = Optional.of(2)
                 expect {
                     expect(nonEmptyValue).isEmptyFun()
                 }.toThrow<AssertionError> {
-                    messageContains("$isDescr: ${EMPTY.getDefault()}")
+                    messageContains("$isDescr: ${DescriptionOptionalAssertion.EMPTY.getDefault()}")
+                }
+            }
+        }
+
+        context("is Present") {
+            it("${isPresent.name} - can perform sub-assertion which holds") {
+                val isPresentValue: Optional<Int> = Optional.of(2)
+                expect(isPresentValue).isPresentFun()
+            }
+
+            it("throws an AssertionError") {
+                expect {
+                    val emptyValue = Optional.empty<Int>()
+                    expect(emptyValue).isPresentFun()
+                }.toThrow<AssertionError> {
+                    messageContains(DescriptionOptionalAssertion.PRESENT.getDefault())
                 }
             }
         }
     }
+
+    //TODO working on the remaining tests for features and nullables
```

Comment:
I will work on those once you confirm I have the right approach for the tests.

Reply:
no need for nullable as outlined above